src/Behat/Gherkin/Loader/CucumberNDJsonAstLoader.php

```diff
@@ -0,0 +1,194 @@
+            self::getTags($featureJson),
+            self::getBackground($featureJson),
+            self::getScenarios($featureJson),
+            $featureJson['keyword'],
+            $featureJson['language'],
+            preg_replace('/(?<=\\.feature).*$/', '', $filePath),
+            $featureJson['location']['line']
+        );
+
+        return $feature;
+    }
+
+    /**
+     * @return string[]
+     */
+    private static function getTags(array $json)
+    {
+        return array_map(
+            static function(array $tag) { return preg_replace('/^@/', '', $tag['name']); },
+            isset($json['tags']) ? $json['tags'] : []
+        );
+    }
+
+    /**
+     * @return ScenarioInterface[]
+     */
+    private static function getScenarios(array $json)
+    {
+
+        return array_values(
+            array_map(
+                static function ($child) {
+
+                    if (isset($child['scenario']['examples'])) {
+                        return new OutlineNode(
+                            $child['scenario']['name'],
+                            self::getTags($child['scenario']),
+                            self::getSteps(isset($child['scenario']['steps']) ? $child['scenario']['steps'] : []),
+                            self::getTables($child['scenario']['examples']),
+                            $child['scenario']['keyword'],
+                            $child['scenario']['location']['line']
+                        );
+                    }
+                    else {
+                        return new ScenarioNode(
+                            $child['scenario']['name'],
+                            self::getTags($child['scenario']),
+                            self::getSteps(isset($child['scenario']['steps']) ? $child['scenario']['steps'] : []),
+                            $child['scenario']['keyword'],
+                            $child['scenario']['location']['line']
+                        );
+                    }
+
+                },
+                array_filter(
+                    $json['children'] ? $json['children'] : [],
+                    static function ($child) {
+                        return isset($child['scenario']);
+                    }
+                )
+            )
+        );
+    }
+
+    /**
+     * @return BackgroundNode|null
+     */
+    private static function getBackground($json)
+    {
+        $backgrounds = array_values(
+            array_map(
+                static function ($child) {
+                    return new BackgroundNode(
+                        $child['background']['name'],
+                        self::getSteps(isset($child['background']['steps']) ? $child['background']['steps'] : []),
+                        $child['background']['keyword'],
+                        $child['background']['location']['line']
+                    );
+                },
+                array_filter(
+                    $json['children'] ? $json['children'] : [],
```

Comment:
should be `isset($json['children']) ? $json['children'] : [],` to support cases where the key is not there (without first triggering a notice about non-existent key, which gets turned into an error on CI)

Reply:
I can't wait for `??` to be available to use in this project :)